### PR TITLE
Gen 5 Random Battles update

### PR DESCRIPTION
**Pull request by livid-washed**
Changes approved by Random Battles staff

https://discord.com/channels/294651453279174656/763039178211721236/985507088937664592

### CODE REVIEW

**Review by adrivrie — APPROVED** [review]

File: data/mods/gen5/random-teams.ts [hunk 1]
```diff
@@ -181,28 +181,41 @@ export class RandomGen5Teams extends RandomGen6Teams {
...
 		case 'bravebird': case 'pluck':
 			return {cull: moves.has('acrobatics') || moves.has('hurricane')};
+		case 'acrobatics':
+			return {cull: !counter.setupType && moves.has('hurricane')};
+		case 'hurricane':
+			return {cull: !!counter.setupType && moves.has('acrobatics')};
 		case 'gigadrain':
-			return {cull: (!counter.setupType && moves.has('leafstorm')) || moves.has('petaldance') || moves.has('powerwhip')};
+			return {cull: (!counter.setupType && moves.has('leafstorm')) ||
+				moves.has('petaldance') || moves.has('powerwhip') || moves.has('leafblade')};
```
Comment (ACakeWearingAHat) [inline data/mods/gen5/random-teams.ts:190]:
```suggestion
				['leafblade', 'petaldance', 'powerwhip'].some(m => moves.has(m));
```

File: data/mods/gen5/random-teams.ts [hunk 2]
```diff
@@ -213,8 +226,12 @@ export class RandomGen5Teams extends RandomGen6Teams {
 			return {cull: moves.has('bodyslam')};
 		case 'healbell':
 			return {cull: !!counter.get('speedsetup') || moves.has('magiccoat')};
-		case 'moonlight': case 'painsplit': case 'recover': case 'roost': case 'softboiled': case 'synthesis':
+		case 'moonlight': case 'painsplit': case 'recover': case 'softboiled': case 'synthesis':
 			return {cull: ['leechseed', 'rest', 'wish'].some(m => moves.has(m))};
+		case 'roost':
```
Comment (ACakeWearingAHat) [inline data/mods/gen5/random-teams.ts:231]: It's unnecessary to separate this; the gliscorCase clause works just fine alongside the other recovery moves and doesn't mess anything up

**Review by ACakeWearingAHat — CHANGES_REQUESTED** [review]

File: data/mods/gen5/random-teams.ts (hunk 2 above)
Comment (livid-washed) [reply data/mods/gen5/random-teams.ts]: You're right, thanks!

**Review by ACakeWearingAHat — APPROVED** [review]

**Review by MathyFurret — APPROVED** [review]
Some regression tests would be nice, but not necessary -- this works

**Pull request merged**